src/GafferScene/Instancer.cpp

```diff
@@ -486,18 +832,71 @@ class Instancer::EngineData : public Data
 		boost::container::flat_map<InternedString, AttributeCreator> m_attributeCreators;
 		MurmurHash m_attributesHash;
 
+		const std::vector< PrototypeContextVariable > m_prototypeContextVariables;
 };
 
 //////////////////////////////////////////////////////////////////////////
 // Instancer
 //////////////////////////////////////////////////////////////////////////
 
+GAFFER_PLUG_DEFINE_TYPE( Instancer::ContextVariablePlug );
+
+Instancer::ContextVariablePlug::ContextVariablePlug( const std::string &name, Direction direction, unsigned flags, bool defaultEnable )
+	:   ValuePlug( name, direction, flags )
+{
+	addChild( new BoolPlug( "enabled", direction, defaultEnable ) );
+	addChild( new StringPlug( "name", direction, "" ) );
+	addChild( new FloatPlug( "quantize", direction, 0.1, 0 ) );
+}
+
+Instancer::ContextVariablePlug::~ContextVariablePlug()
+{
+}
+
+bool Instancer::ContextVariablePlug::acceptsChild( const GraphComponent *potentialChild ) const
+{
+	return children().size() < 3;
+}
+
+Gaffer::PlugPtr Instancer::ContextVariablePlug::createCounterpart( const std::string &name, Direction direction ) const
+{
+	return new Instancer::ContextVariablePlug( name, direction, getFlags() );
```

Comment:
Is `enabledPlug()->defaultValue()` missing here?

Reply:
Addressed.